### ISSUE TITLE
Unable to get a page to load

### ISSUE THREAD

**Issue by kriscii** [hay-kot/mealie#194]
Using the docker-compose file in the docs and pulling from docker hub on a new clean install just getting 404 not found when trying to access. Going to /docs does load up. Tried the other images on Docker hub and same problem. First time users so it could be me but chose to go the docker hub way to make it foolproof but obviously not?

**Comment by hay-kot** [comment]
The link might nor be complete in the docs. Try /api/docs

**Comment by hay-kot** [comment]
Can you post your docker compose?

**Comment by kriscii** [comment]
I was expecting the UI on http://ipaddr:9900/  I get 404 0n this. 


I get the docs on http://ipaddr:9900/docs

The compose file is 
`version: "3.1"
services:
  mealie:
    container_name: mealie
    image: hkotel/mealie:latest
    restart: always
    ports:
      - 9000:9000
    environment:
      db_type: sqlite
      TZ: Europe/London
    volumes:
      - ./mealie/data/:/app/data
`
The logs say:
{"level":"info","ts":1614788663.086382,"msg":"using provided configuration","config_file":"./Caddyfile","config_adapter":""},
run: loading initial config: loading new config: starting caddy administration endpoint: listen tcp 192.168.8.34:2019: bind: cannot assign requested address,
start: caddy process exited with error: exit status 1,
03-Mar-21 16:24:24 INFO: Adding job tentatively -- it will be properly scheduled when the scheduler starts,
03-Mar-21 16:24:24 INFO: ----INIT SCHEDULE OBJECT-----,
03-Mar-21 16:24:24 INFO: Adding job tentatively -- it will be properly scheduled when the scheduler starts,
03-Mar-21 16:24:24 INFO: New Function Scheduled,
03-Mar-21 16:24:24 INFO: None,
03-Mar-21 16:24:24 INFO: Adding job tentatively -- it will be properly scheduled when the scheduler starts,
03-Mar-21 16:24:24 INFO: New Function Scheduled,
03-Mar-21 16:24:24 INFO: None,
03-Mar-21 16:24:24 INFO: Added job "update_webhook_schedule" to job store "default",
03-Mar-21 16:24:24 INFO: Added job "backups" to job store "default",
03-Mar-21 16:24:24 INFO: Added job "webhooks" to job store "default",
03-Mar-21 16:24:24 INFO: Scheduler started,
INFO:     Started server process [28],
03-Mar-21 16:24:24 INFO: Started server process [28],
INFO:     Waiting for application startup.,
03-Mar-21 16:24:24 INFO: Waiting for application startup.,
INFO:     Application startup complete.,
03-Mar-21 16:24:24 INFO: Application startup complete.,
INFO:     Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit),
03-Mar-21 16:24:24 INFO: Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit),
Pending jobs:,
    update_webhook_schedule (trigger: interval[0:15:00], pending),
    backups (trigger: cron[hour='0', minute='0'], pending),
Pending jobs:,
    update_webhook_schedule (trigger: interval[0:15:00], pending),
    backups (trigger: cron[hour='0', minute='0'], pending),
    webhooks (trigger: cron[hour='0', minute='0'], pending),
INFO:     192.168.8.70:56341 - "GET / HTTP/1.1" 404 Not Found,
INFO:     192.168.8.70:56344 - "GET /docs HTTP/1.1" 200 OK,
INFO:     192.168.8.70:56344 - "GET /openapi.json HTTP/1.1" 200 OK,

**Comment by hay-kot** [comment]
You need to change your port mapping to 9000:80

**Comment by kriscii** [comment]
I did that originally but don't even get a response from that so I changed it to 9000:9000 because the logs said they were listening on 9000. 

I have changed it to
    ports:
      - 9000:80
 but I still get no response the log shows nothing other than listening on 9000:

INFO:     Started server process [27]
03-Mar-21 16:39:36 INFO: Started server process [27]
INFO:     Waiting for application startup.
03-Mar-21 16:39:36 INFO: Waiting for application startup.
INFO:     Application startup complete.
03-Mar-21 16:39:36 INFO: Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)
03-Mar-21 16:39:36 INFO: Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)

**Comment by hay-kot** [comment]
The API listens on Port 9000, the Caddy webserver proxies requests on port 80 to either to static site files or the API depending on the url. I'm unable to replicate the issue on my system. The below docker-compose worked fine for me. I suggest deleting all your local docker images and re-pulling the latest image to see if that does anything. Other than that, I'm not sure what's going on. What hardware/OS are you deploying on? 

```yml
version: "3.1"
services:
  mealie:
    container_name: mealie
    image: hkotel/mealie:latest
    restart: always
    ports:
      - 9000:80
    environment:
      db_type: sqlite
      TZ: Europe/London
    volumes:
      - ./mealie/data/:/app/data

```

**Comment by kriscii** [comment]
If you look above in the log Caddy is dying at startup.  Is the 192.168.8.34 in the log excerpt below hardcoded within the container as my network is 192.168.8.0/24 as it could be causing the issue. 

{"level":"info","ts":1614788663.086382,"msg":"using provided configuration","config_file":"./Caddyfile","config_adapter":""},
run: loading initial config: loading new config: starting caddy administration endpoint: listen tcp 192.168.8.34:2019: bind: cannot assign requested address,
start: caddy process exited with error: exit status 1,

It is running on a Intel NUC with Ubuntu 18.04.4 LTS. I tried all older containers on Docker Hub and it didn't work.

**Comment by hay-kot** [comment]
On startup Caddy tries to resolve localhosts IP addresses and during that process it comes up with 192.168.8.34, which is maybe the Ubuntu 18.04 machines address? It's possible that somehow docker is resolving localhost to a different address. I've never seen this issues before. In the future I can turn off the admin endpoint in the CaddyFile so this issue doesn't come up. As for the issue you're experiencing now, I'm unsure of a resolution. I think this has more to do with your Docker setup than with Mealie. To test, you could clone the repo, change the CaddyFiles contents to the below and build from source. 

```
{
	auto_https off
        admin off
}

:80 {
  @proxied path /api/* /docs /openapi.json

  root * /app/dist
  encode gzip
  uri strip_suffix /
  
  handle @proxied {
    reverse_proxy http://127.0.0.1:9000 
  }

  handle {
    try_files {path}.html {path} /
    file_server 
  }

}
```

Relevant Caddy Issue
https://github.com/caddyserver/caddy/issues/3440

**Comment by kriscii** [comment]
Which directory is the CaddyFile in I could just map that file out from Container

**Comment by hay-kot** [comment]
from the Dockerfile, the Caddyfile is copied to the /app directory.

```
COPY ./Caddyfile /app
```

**Comment by kriscii** [comment]
I turned admin off and it allows access now.

I checked the ip addresses in the container and there are no 192.168.8. addresses. This is what I get from the ip command:
/app # ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
113: eth0@if114: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue state UP 
    link/ether 02:42:c0:a8:40:02 brd ff:ff:ff:ff:ff:ff
    inet 192.168.64.2/20 brd 192.168.79.255 scope global eth0
       valid_lft forever preferred_lft forever

**Comment by hay-kot** [comment]
Weird. I've added the `admin off` directive to the Caddyfile for the next release, so it shouldn't be a problem moving forward. I appreciate you helping solve the problem!

**Comment by hay-kot** [comment]
Fixed in [v0.4.0](https://hay-kot.github.io/mealie/changelog/v0.4.0/)
